src/sock.c

```diff
@@ -99,6 +100,17 @@ sock_t sock_connect(const char * const host, const unsigned int port)
         if (sock < 0)
             continue;
 
+	/* turn on keepalive */
```

Comment:
Optional feature looks good, but what about timeout value? It have to be configurable.

As for enabling keepalives by default - it's not a good idea, like you already said.

TCP Keepalives is better than XMPP pings in many ways. First, they don't require server to support them. It's just a TCP layer level basic functionality, just sending zero-length packet with ack flag set and getting acknowledgement (also zero-length ack) from peer, OS will take care of it. Second, TCP keepalives takes minimum possible traffic. They are just zero-length packets with ack bit set.

Try them, may be you'll find them interesting. :)